doc/kernel/format_specifications.rdoc

```diff
@@ -0,0 +1,230 @@
+== Format Specifications
+
+Several Ruby core classes have instance method +printf+ or +sprintf+:
+
+- ARGF#printf
+- IO#printf
+- Kernel#printf
+- Kernel#sprintf
+
+Each of these methods takes:
+
+- Argument +format_string+, which has zero or more
+  embedded _format_ _specifications_ (see below).
+- Arguments <tt>*arguments</tt>, which are zero or more objects to be formatted.
+
+Each of these methods returns the string resulting from replacing each
```

Comment:
Only `Kernel#sprintf` returns the string, the other methods print the string to the related IO/stdout, then return nil.